docs/features/communication/abi_compatible_data_types/index.rst

```diff
@@ -0,0 +1,491 @@
+* Support all ABI-compatible data types previously defined.
+* Include versioning to manage schema evolution and compatibility.
+* Allow easy generation and parsing by tooling in both C++ and Rust.
+
+Workflows
+^^^^^^^^^
+
+Two potential workflows are considered for creating type descriptions:
+
+**Description-first Workflow**: The type description is defined independently (e.g., via a schema or domain-specific language). The C++ and Rust type definitions are then generated based on this description as part of the application build process.
+
+**Definition-first Workflow**: Existing type definitions in Rust or C++ are the source of truth, and the corresponding type description is generated during the build process via compiler tooling.
+
+Both workflows are valid, and the final decision is deferred pending further feasibility analysis.
+
+Type Description Format
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The format of the type description shall explicitly support versioning to allow schema evolution and backward compatibility. It must accommodate all data types described earlier in the ABI compatibility section. It should be simple, human-readable, and easily machine-parsable.
+
+The choice of serialization format is left open but may include RON, JSON5, or a custom DSL, based on readability, tooling support, and maintainability.
+
+
+.. Backwards Compatibility
+.. =======================
+
+
+Security Impact
+===============
+
+.. note::
+
+   This section does not replace a formal security impact analysis. This only guides the design thoughts behind security related topics and is to be understood as a starting point for later workflows.
+
+No assumed impact on security.
+
+
+Safety Impact
+=============
+
+.. note::
+
+   This section does not replace a formal safety impact analysis. This only guides the design thoughts behind safety related topics and is to be understood as a starting point for later workflows.
+
+All functionality shall be available to applications rated up to ASIL-B.
+
+
+.. License Impact
+.. ==============
+
+
+.. How to Teach This
+.. ==================
+
+
+Rejected Ideas
+==============
+
+Reflection
+----------
+
+Reflection, in this context, is the ability to inspect data at runtime even if its structure is not or not fully known at compile time.
+Benefits of reflection include being able to translate recorded data into a human-readable format (e.g., JSON or CSV) without having to know the type definitions at compile time; this enables general-purpose data recording and transformation tools.
+
+This ability requires some form of *type description* being available at runtime, so that a sequence of bytes can be interpreted as a data structure.
+There are two primary approaches to achieve this goal:
+
+* *inline type descriptions*, which precede each instance of every type, and
+* *external type descriptions*, which are stored separately from the data.
```

Comment:
I feel like you argue reflection a lot in the type description sections above. But here you reject the idea.
Could you rephrase the sections on type description to clear this up?